main.py

```diff
@@ -0,0 +1,254 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy import create_engine, Column, Integer, String, Numeric, ForeignKey, DateTime
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker, Session
+from pydantic import BaseModel
+from datetime import datetime
+from typing import List
+from sqlalchemy.exc import SQLAlchemyError
+from passlib.context import CryptContext
+from sqlalchemy.orm import declarative_base
+
+
+# Создание подключения к базе данных
+SQLALCHEMY_DATABASE_URL = "sqlite:///./shop.db"
+engine = create_engine(SQLALCHEMY_DATABASE_URL)
+SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+Base = declarative_base()
+
+# Определение моделей SQLAlchemy
+class User(Base):
+    __tablename__ = "users"
+    id = Column(Integer, primary_key=True, index=True)
+    name = Column(String)
+    surname = Column(String)
+    email = Column(String, unique=True)
+    password = Column(String)
+
+class Product(Base):
+    __tablename__ = "products"
+    id = Column(Integer, primary_key=True, index=True)
+    title = Column(String)
+    description = Column(String)
+    price = Column(Numeric(10, 2))
+
+class Order(Base):
+    __tablename__ = "orders"
+    id = Column(Integer, primary_key=True, index=True)
+    user_id = Column(Integer, ForeignKey("users.id"))
+    product_id = Column(Integer, ForeignKey("products.id"))
+    order_date = Column(DateTime)
+    status = Column(String)
+
+# Создание всех таблиц в базе данных
+Base.metadata.create_all(bind=engine)
+
+# Pydantic модели для валидации данных
+class UserCreate(BaseModel):
+    name: str
+    surname: str
+    email: str
+    password: str
+
+class ProductCreate(BaseModel):
+    title: str
+    description: str
+    price: float
+
+class OrderCreate(BaseModel):
+    user_id: int
+    product_id: int
+    order_date: datetime
+    status: str
+
+class UserRead(BaseModel):
+    id: int
+    name: str
+    surname: str
+    email: str
+
+    class Config:
+        from_attributes = True
+
+class ProductRead(BaseModel):
+    id: int
+    title: str
+    description: str
+    price: float
+
+    class Config:
+        from_attributes = True
```

Comment:
Модели очень похожи, а значит можно создать одну из них, наследуясь от другой.
```suggestion
class ProductRead(ProductCreate):
    id: int

    class Config:
        from_attributes = True
```